include/deal.II/base/tensor.h

```diff
@@ -3048,54 +3037,6 @@ linfty_norm(const Tensor<2, dim, Number> &t)
 
 /** @} */
 
-
-#ifndef DOXYGEN
-
-
-#  ifdef DEAL_II_ADOLC_WITH_ADVANCED_BRANCHING
-
-// Specialization of functions for ADOL-C number types when
-// the advanced branching feature is used
-template <int dim>
-inline adouble
-l1_norm(const Tensor<2, dim, adouble> &t)
-{
-  adouble max = internal::NumberType<adouble>::value(0.0);
-  for (unsigned int j = 0; j < dim; ++j)
-    {
-      adouble sum = internal::NumberType<adouble>::value(0.0);
-      for (unsigned int i = 0; i < dim; ++i)
-        sum += std::fabs(t[i][j]);
-
-      condassign(max, (sum > max), sum, max);
```

Comment:
@drwells For these taped types, one needs to use the `condassign()` function to switch between branches. An `if` statement won't work if the evaluation point changes which branch is taken.

Reply:
I'll write a test so that I understand this better.

Reply:
I had the ADOL-C manual in hand when decomposing and writing these functions, so whenever you come across an unusual implementation then I'd suggest consulting the manual. The requirements of the taping algorithm often dictates how code with branches has to be written in order to be taped AD compatible. It would not surprise me if there is not a test for every single specialised function -- the implementation was probably "best effort".